### PR TITLE
Support missing signature modifiers

### PR DESCRIPTION
**Pull request by prolativ**
* Support missing signature modifiers (open + private/protected with scope)
* Extend test framework

### CODE REVIEW

**Comment by prolativ** [comment]
Closes #21

File: README.md [hunk 1]
```diff
@@ -66,6 +66,52 @@ You can also find the result of building the same sites for latest `master` at:
...
+
+class Bar
+{
+   //...
+}
+```
+
+Otherwise the implementation would be treated as a part of the signature.
+
+`MultipleFileTest` and `SingleFileTest` test classes allow you to specify which kinds of signatures from a source file (depending on the keyword used to declare them like `def`, `class`, `object` etc.) are checked and which are ignored in a test.
```
Comment (abgruszecki) [inline README.md]: Can we move this paragraph to be the first in this section and reword it a bit to explicitly say that 1) a test is, in the end, defined by extending either `MultipleFileTest` or its subclass, `SingleFileTest` and that 2) this section in the readme might be out-of-date and if that seems to be the case, the sources of those classes should be read?

File: src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala [hunk 2]
```diff
@@ -61,9 +61,14 @@ trait ClassLikeSupport:
       )
 
   extension (c: ClassDef):
+      def nonprivateMembers = c.body.filterNot(_.symbol.isHiddenByVisibility)
```
Comment (abgruszecki) [inline src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala]: The name suggests that only private members are filtered out, can we instead name this `membersToDocument` (as that seems to be the intention here)?

File: src/main/scala/tests/visibility.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    protected def protectedMethod: Int
+      = ???
+
+    protected[tests] def protectedInOuterPackageMethod: Int //unexpected
+      = ???
+
+    protected[visibility] def protectedInInnerPackageMethod: Int //unexpected
+      = ???
+
+    protected[InClassVisibility] def protectedInClassClassMethod: Int //unexpected
```
Comment (abgruszecki) [inline src/main/scala/tests/visibility.scala]: We actually want to document at least some class-local definitions; in particular, this one. `protected[InClassVisibility]` inside `InClassVisibility` is the same as `protected`, and we show those. In principle, I think we want to document class-local definitions if the class they're local in is itself documented, but if that's too difficult to implement, we can just always document them.

**Review by abgruszecki — COMMENTED** [review]
Overall LGTM - adding the `//unexpected` marker was a nice touch, documenting the testing framework in the readme is a very good idea as well.

File: src/main/scala/tests/visibility.scala (hunk 3 above)
Comment (prolativ) [reply src/main/scala/tests/visibility.scala]: And should they all be displayed as `protected` inside the documentation? Or should we try to preserve the original definition, i.e. `protected` or `protected[EnclosingClass]`?

Comment (abgruszecki) [reply src/main/scala/tests/visibility.scala]: It's fine to display them as `protected[EnclosingClass]` - this is what the user entered and trying to special-case how it is displayed isn't especially important.

File: README.md (hunk 1 above)
Comment (prolativ) [reply README.md]: done

File: src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala (hunk 2 above)
Comment (prolativ) [reply src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala]: done

**Review by romanowski — APPROVED** [review]

**Pull request merged**